File: 0007-reverse-integer/0007-reverse-integer.cpp
class Solution {
public:
    int reverse(int x) {
        int d=0;
        long s=0;
        while(x) {
            d=x%10;
            s=(s*10)+d;
            x/=10;
        }
        if(s>INT_MIN and s<INT_MAX)
        return s;
        else
        return 0;
    }
};

File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        int res = 0;
        while(x != 0) {
            int d = x % 10;
            int s = res * 10 + d;
            if((s - d) / 10 != res)
            return 0;
            res = s;
            x /= 10;
        }
        return res;
    }
}